bundler/spec/support/artifice/endpoint_mirror_source.rb

```diff
@@ -4,7 +4,7 @@
 
 class EndpointMirrorSource < Endpoint
   get "/gems/:id" do
-    if request.env["HTTP_X_GEMFILE_SOURCE"] == "https://server.example.org/"
+    if request.env["HTTP_X_GEMFILE_SOURCE"] == "https://server.example.org/" && request.env["HTTP_USER_AGENT"].start_with?("bundler")
```

Comment:
This is tied to `spec/realworld/gemfile_source_header_spec.rb` spec, right? And you have expanded it to also check that a proper Bundler user agent is set. It's still not a very straighforward spec, but we I would add this to the spec description: 

https://github.com/rubygems/rubygems/blob/d45b23d886e7eaa351d7214d51e595d505e8f0e8/bundler/spec/realworld/gemfile_source_header_spec.rb#L20

and maybe also rename the file.